fe/src/main/java/org/apache/doris/transaction/GlobalTransactionMgr.java

```diff
@@ -447,6 +447,114 @@ public void commitTransaction(long dbId, long transactionId, List<TabletCommitIn
         // 6. update nextVersion because of the failure of persistent transaction resulting in error version
         updateCatalogAfterCommitted(transactionState, db);
         LOG.info("transaction:[{}] successfully committed", transactionState);
+
+        // 7. try publishing version as soon as possible
+        if (couldPublishTransaction(transactionState)) {
+            publishTransaction(transactionState, totalInvolvedBackends);
+        } else {
+            // add publish version tasks. set task to null as a placeholder.
+            // tasks will be created when publishing version
+            for (long backendId : totalInvolvedBackends) {
+                transactionState.addPublishVersionTask(backendId, null);
+            }
+        }
+    }
+
+    // the publishing transaction should meet:
+    // 1. the transaction state is COMMITTED
+    // 2. the partition's visible version plus one should equal transaction version.
+    public boolean couldPublishTransaction(TransactionState transactionState) {
```

Comment:
I think there needs to be a synchronization method to ensure that only one thread can check if a transaction can be published. Otherwise the transaction may be checked multiple times, affecting performance. 

And then the `publishTransaction()` method also needs a sync method to ensure that only one thread can try to send the publish task of the transaction. The `hasSendTask()` method of transactionState is not enough. It is not a mutually exclusive variable.

For example, adding a memtable in transactionState named publish object:
```
byte[] publishLock = new byte[0];
...

public boolean couldPublishTransaction() {
    synchronized(publishLock) {
         // do the check...
    }
}

public void publishTransaction() {
    synchronized(publishLock) {
        if (hasSendTask()) { return; }
        // send task
        sendTask = true;
     }
     return;
}

```

Reply:
I agree with you, but which is a little complicated.  I change back, and only let `PublishVersionDaemon` handle `sendPublishVersionTask`.